### PR TITLE
feat: bulk add a number of otel instrumentations

### PR DESCRIPTION
**Pull request by trentm**
```
@opentelemetry/instrumentation-connect
@opentelemetry/instrumentation-cucumber
@opentelemetry/instrumentation-dataloader
@opentelemetry/instrumentation-dns
@opentelemetry/instrumentation-generic-pool
@opentelemetry/instrumentation-knex
@opentelemetry/instrumentation-koa
@opentelemetry/instrumentation-lru-memoizer
@opentelemetry/instrumentation-memcached
@opentelemetry/instrumentation-nestjs-core
@opentelemetry/instrumentation-net
@opentelemetry/instrumentation-restify
@opentelemetry/instrumentation-router
@opentelemetry/instrumentation-socket.io
```

---

Notably no tests have been added for these. The minor justification being that these are instrumentations that seem to very rarely change upstream. We can come back and add tests for these as/if needed.

### CODE REVIEW

**Comment by trentm** [comment]
Draft for now. I need to work through tests and ignore instr-dns and instr-net spans that are throwing off assertions.
It would be nice to have `OTEL_NODE_DISABLED_INSTRUMENTATIONS` that auto-instrumentations-node is getting. Then the test fixtures coudl run with `OTEL_NODE_DISABLED_INSTRUMENTATIONS: 'dns,net'`.

File: packages/opentelemetry-node/lib/instrumentations.js [hunk 1]
```diff
@@ -24,58 +24,68 @@
...
- *  "@opentelemetry/instrumentation-pg": import('@opentelemetry/instrumentation-pg').PgInstrumentation | InstrumentationFactory
- *  "@opentelemetry/instrumentation-pino": import('@opentelemetry/instrumentation-pino').PinoInstrumentation | InstrumentationFactory
- *  "@opentelemetry/instrumentation-redis-4": import('@opentelemetry/instrumentation-redis-4').RedisInstrumentation | InstrumentationFactory,
- *  "@opentelemetry/instrumentation-tedious": import('@opentelemetry/instrumentation-tedious').TediousInstrumentation | InstrumentationFactory,
+ *  "@opentelemetry/instrumentation-knex": import('@opentelemetry/instrumentation-knex').KnexInstrumentationConfig | InstrumentationFactory,
+ *  "@opentelemetry/instrumentation-koa": import('@opentelemetry/instrumentation-koa').KoaInstrumentationConfig | InstrumentationFactory,
+ *  "@opentelemetry/instrumentation-lru-memoizer": import('@opentelemetry/instrumentation').InstrumentationConfig | InstrumentationFactory,
+ *  "@opentelemetry/instrumentation-memcached": import('@opentelemetry/instrumentation').InstrumentationConfig | InstrumentationFactory,
```
Comment (david-luna) [inline packages/opentelemetry-node/lib/instrumentations.js:43]: this instrumentation is exporting a config interface with the same name but actually it has an extra property.

ref: https://github.com/open-telemetry/opentelemetry-js-contrib/blob/2dc2f72e5a86aa9e1220a26f78878070acde8424/plugins/node/opentelemetry-instrumentation-memcached/src/types.ts#L19

```suggestion
 *  "@opentelemetry/instrumentation-memcached": import('@opentelemetry/instrumentation-memcached').InstrumentationConfig | InstrumentationFactory,
```

**Comment by david-luna** [comment]
> It would be nice to have OTEL_NODE_DISABLED_INSTRUMENTATIONS that auto-instrumentations-node is getting. Then the test fixtures coudl run with OTEL_NODE_DISABLED_INSTRUMENTATIONS: 'dns,net'.

Do you plan to do it in this PR or you prefer to have a dedicated issue to honor the env var `OTEL_NODE_DISABLED_INSTRUMENTATIONS` ?

**Review by david-luna — CHANGES_REQUESTED** [review]

**Comment by trentm** [comment]
> Do you plan to do it in this PR or you prefer to have a dedicated issue to honor the env var `OTEL_NODE_DISABLED_INSTRUMENTATIONS` ?

Not for this PR, no. I think when auto-instrumentations-node has this merged we should consider supporting the same envvar.  And then when we do we could switch the tests cases to use that instead of the `filterOut...()` hack I added here.

**Review by david-luna — APPROVED** [review]

**Pull request merged**